airflow/models/taskinstance.py

```diff
@@ -2237,27 +2237,40 @@ def xcom_push(
         :param value: A value for the XCom. The value is pickled and stored
             in the database.
         :type value: any picklable object
-        :param execution_date: if provided, the XCom will not be visible until
-            this date. This can be used, for example, to send a message to a
-            task on a future date without it being immediately visible.
+        :param execution_date: Deprecated parameter.
         :type execution_date: datetime
         :param session: Sqlalchemy ORM Session
         :type session: Session
         """
-        self_execution_date = self.get_dagrun(session).execution_date
-        if execution_date and execution_date < self_execution_date:
-            raise ValueError(
-                f'execution_date can not be in the past (current execution_date is '
-                f'{self_execution_date}; received {execution_date})'
-            )
+        run_id = None
+        if execution_date:
+            self_execution_date = self.get_dagrun(session).execution_date
+            if execution_date < self_execution_date:
+                raise ValueError(
+                    f'execution_date can not be in the past (current execution_date is '
+                    f'{self_execution_date}; received {execution_date})'
+                )
+            elif execution_date != self_execution_date:
+                warnings.warn(
+                    "Passing `execution_date` parameter to xcom_push is deprecated.",
+                    DeprecationWarning,
+                    stacklevel=3,
+                )
```

Comment:
This could possibly be extended to "passing a date at all"

Reply:
I made this unconditionally warn instead.